### DISCUSSION TITLE
Failed to create Salesforce Case

### DISCUSSION THREAD

**Discussion by RMinayaCar** [amazon-connect/amazon-connect-salesforce-scv#93] [Q&A] [answered]
Hello Everyone,

I'm currently trying to create a voicemails into Salesforce Cases, but I'm getting the next error in the VMXPackager: Failed to create Salesforce case. Full log below:

START RequestId: 95442d2f-9fda-4e4b-bae1-42b19aeedb1e Version: $LATEST
[ERROR]	2022-04-06T16:36:02.657Z	95442d2f-9fda-4e4b-bae1-42b19aeedb1e	
[ERROR]	2022-04-06T16:36:02.657Z	95442d2f-9fda-4e4b-bae1-42b19aeedb1e	Record 1 Result: Failed to create Salesforce Case
END RequestId: 95442d2f-9fda-4e4b-bae1-42b19aeedb1e
REPORT RequestId: 95442d2f-9fda-4e4b-bae1-42b19aeedb1e	Duration: 8635.85 ms	Billed Duration: 8636 ms	Memory Size: 128 MB	Max Memory Used: 100 MB	Init Duration: 459.61 ms

Request ID
95442d2f-9fda-4e4b-bae1-42b19aeedb1e

Is there anything that I might be missing, please help!

**Comment by dougjaso** [comment]
Questions:

- Are you using the [VMX2 version of voicemail express](https://github.com/amazon-connect/amazon-connect-salesforce-scv/tree/master/Solutions/VMX2-VoicemailExpress)? 
- When you setup [Salesforce Config](https://github.com/amazon-connect/amazon-connect-salesforce-scv/tree/master/Common/AWSSCV-SalesforceConfig), did your validation test work correctly?
- Can you validate that the [custom fields](https://github.com/amazon-connect/amazon-connect-salesforce-scv/blob/master/Solutions/VMX2-VoicemailExpress/Docs/vmx_prerequistes.md#create-a-custom-field-for-the-contact-attributes) are all created and that the API Name was used in the deployment?

**Reply by dougjaso** [reply]
Also, can you set logging level to DEBUG and run another test so you get more info from log?

**Reply by RMinayaCar** [reply]
I was using the old VMX solution and it was giving me the same sa me error as this one, all the validation worked previously, I assigned all the custom fields with the correct API names

Im updating the stack to DEBUG level, sharing the results shortly

**Reply by RMinayaCar** [reply]
*****REDACTED*****

**Reply by dougjaso** [reply]
Can you reach out to me at dougjaso@amazon.com so we can keep your account details private? Thanks

**Reply by RMinayaCar** [reply]
Will do

**Reply by RMinayaCar** [reply]
Just emailed you

**Comment by dougjaso** [comment, marked as answer]
Misconfiguration of Consumer key. Spoke directly to resolve.

**Comment by RMinayaCar** [comment]
Yup that was it, is working now Jason!

Thank you so much for your help

**Comment by dougjaso** [comment]
Awesome. Glad to hear it

-----------
Jason Douglas
Principal Solutions Architect
Amazon Connect

628.336.1973 – mobile
415.262.9741 – office

[signature_959016464]<http://www.aws.amazon.com/>


From: RMinayaCar ***@***.***>
Reply-To: amazon-connect/amazon-connect-salesforce-scv ***@***.***>
Date: Wednesday, April 6, 2022 at 12:35 PM
To: amazon-connect/amazon-connect-salesforce-scv ***@***.***>
Cc: "Douglas, Jason" ***@***.***>, Comment ***@***.***>
Subject: Re: [amazon-connect/amazon-connect-salesforce-scv] Failed to create Salesforce Case (Discussion #93)

Yup that was it, is working now Jason!

Thank you so much for your help

**Comment by RMinayaCar** [comment]
I'm having another issue now:

[ERROR] Runtime.UserCodeSyntaxError: Syntax error in module 'vmx_packager':
invalid syntax (sub_salesforce_case.py, line 69)
Traceback (most recent call last):
  File "/var/task/sub_salesforce_case.py" Line 69
                    'RecordTypeId' = '012R0000003Hrx9IAC',


Is that because of the structure of the code, maybe I wrote it wrong?
